cardano-api/golden/Test/Golden/ErrorsSpec.hs

```diff
@@ -295,3 +302,15 @@ spec = describe "Test.Golden.Errors" $ do
   it "TxMetadataRangeError" $ H.requireTest $ do
     flip H.diffVsGoldenFile "golden/files/errors/TxMetadataRangeError/TxMetadataJsonToplevelNotMap.txt"
       $ displayError $ TxMetadataBytesTooLong 0
+
+testErrorMessage :: (HasCallStack, MonadIO m, MonadTest m, Data p , Error p) => p -> m ()
+testErrorMessage err = withFrozenCallStack $ do
+  let typeName = show $ typeOf err
+      constructorName = show $ toConstr err
+  H.note_ "Incorrect error message in golden file"
+  displayError err `H.diffVsGoldenFile` ("golden/files/errors/" <> typeName <> "/" <> constructorName <> ".txt")
+
+deriving instance Typeable ProtocolParametersConversionError
+deriving instance Data ProtocolParametersConversionError
+deriving instance Data Plutus.CostModelApplyError
```

Comment:
I wonder if Plutus team would be amenable to adding this instance?

Reply:
I can make PR and we will see. Also they seem to be slow with releases, so we would have to live with an orphan for some period.